### PR TITLE
v2.x: oshmem: Align OSHMEM API with spec v1.3 (alltoall)

### PR DESCRIPTION
**Pull request by igor-ivanov**
trunk: https://github.com/open-mpi/ompi/pull/1478

:bot:assign: @hppritcha
:bot:label:enhancement
:bot:milestone:v2.0.0

open-mpi/ompi@50906b34b311811e771d4678a42236e3ef62ab91
open-mpi/ompi@bd6eaac56139f6997b0b279405779c4746ff51fd
open-mpi/ompi@3e1e131744fb88b3af559b02247d8074ecaf1220
open-mpi/ompi@9825157fc424d7872953d1acfbbf0d052537d0c3
open-mpi/ompi@1bed5d8aee08cf71aba1154d0823b92919bfeb7c

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1464/ for details.

**Comment by hppritcha** [comment]
This isn't a bug fix. Only bug fixes going in to 2.0.0 now.  Moving to 2.1 milestone.

**Comment by hppritcha** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1469/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1470/ for details.

File: oshmem/include/pshmem.h [hunk 1]
```diff
@@ -245,6 +245,10 @@ OSHMEM_DECLSPEC void pshmem_collect32(void *target, const void *source, size_t n
 OSHMEM_DECLSPEC void pshmem_collect64(void *target, const void *source, size_t nlong, int PE_start, int logPE_stride, int PE_size, long *pSync);
 OSHMEM_DECLSPEC void pshmem_fcollect32(void *target, const void *source, size_t nlong, int PE_start, int logPE_stride, int PE_size, long *pSync);
 OSHMEM_DECLSPEC void pshmem_fcollect64(void *target, const void *source, size_t nlong, int PE_start, int logPE_stride, int PE_size, long *pSync);
+OSHMEM_DECLSPEC void pshmem_alltoall32(void *target, const void *source, size_t nelems, int PE_start, int logPE_stride, int PE_size, long *pSync);
```
Comment (ggouaillardet) [inline oshmem/include/pshmem.h:248]: i did not see that in master, so i committed it in open-mpi/ompi@cb84f582b282232a412816686986fdf9d05b9869

Comment (igor-ivanov) [inline oshmem/include/pshmem.h:248]: @ggouaillardet you are right. I noticed it too yesterday but forgot to push. Thank you.

**Comment by artpol84** [comment]
bot:mellanox:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/2055/ for details.

**Comment by hppritcha** [comment]
@artpol84 as a sanity check, are there any dependencies of this PR on other oshmem PR's going in to 2.1?

**Comment by artpol84** [comment]
@hppritcha - shouldn't be as nothing was indicated in the title/comments of this PR.

**Pull request merged**